Mathlib/Topology/Homeomorph.lean

```diff
@@ -1048,3 +1048,33 @@ lemma IsHomeomorph.pi_map {ι : Type*} {X Y : ι → Type*} [∀ i, TopologicalS
     [∀ i, TopologicalSpace (Y i)] {f : (i : ι) → X i → Y i} (h : ∀ i, IsHomeomorph (f i)) :
     IsHomeomorph (fun (x : ∀ i, X i) i ↦ f i (x i)) :=
   (Homeomorph.piCongrRight fun i ↦ (h i).homeomorph (f i)).isHomeomorph
+
+/-- `HomeomorphClass F A B` states that `F` is a type of two-sided continuous morphisms.-/
+class HomeomorphClass (F : Type*) (A B : outParam Type*)
+    [TopologicalSpace A] [TopologicalSpace B] [h : EquivLike F A B] : Prop where
+  continuous_toFun : ∀ (f : F), Continuous (h.coe f)
+  continuous_invFun : ∀ (f : F), Continuous (h.inv f)
+
+namespace HomeomorphClass
+
+variable {F α β : Type*} [TopologicalSpace α] [TopologicalSpace β] [EquivLike F α β]
+
+/-- Turn an element of a type `F` satisfying `HomeomorphClass F α β` into an actual
+`Homeomorph`. This is declared as the default coercion from `F` to `α ≃ₜ β`. -/
+def toHomeomorph [h : HomeomorphClass F α β] (f : F) : α ≃ₜ β :={
+  (f : α ≃ β) with
+  continuous_toFun := h.continuous_toFun f
+  continuous_invFun := h.continuous_invFun f
+  }
+
+instance [HomeomorphClass F α β] : CoeTC F (α ≃ₜ β) :=
+  ⟨HomeomorphClass.toHomeomorph⟩
+
+theorem toHomeomorph_injective [HomeomorphClass F α β] : Function.Injective ((↑) : F → α ≃ₜ β) :=
+  fun _ _ e ↦ DFunLike.ext _ _ fun a ↦ congr_arg (fun e : α ≃ₜ β ↦ e.toFun a) e
```

Comment:
I don't think we have this for other morphism classes. I suspect it's not necessary as we can just apply the coercion to functions and then use `DFunLike.coe_injective` (probably after `simp`)

Reply:
I copied this lemma from `MulEquivClass`, I didn't think carefully whether it is useful.

Reply:
Okay, it's fine to leave it then; perhaps it's useful. We don't have this lemma for most other coercions to morphisms though (which is why I didn't think it was likely to be useful).